### PR TITLE
fix: serialise AnimatePresence mode="wait"

### PR DESCRIPTION
**Pull request by JonathonRP**
## The bug

`AnimatePresence mode="wait"` did not serialise. The incoming child mounted and took up layout
while the outgoing child was still exiting, so both were laid out at once for the entire exit.

## Root cause

The incoming child was meant to be held out of flow by `motionEnterIntro`:

```ts
const delay = context?.mode === 'wait' ? context.remaining() : 0;
return { delay, duration: 0, css: delay > 0 ? (t) => (t < 1 ? 'display: none' : '') : undefined };
```

`remaining()` is fed by `context.reserve(duration)`, which the *outgoing* child's outro calls.
But Svelte builds the incoming keyed block **before** tearing the outgoing one down, so the intro
almost always read `0` — which meant `css` was `undefined` and no guard was emitted at all.

A DOM dump taken mid-exit confirmed it: two `.page` nodes, both `display: block`, and **zero**
Svelte-generated `<style>` tags, i.e. the CSS transition never existed.

Interestingly the *animation* half of wait mode already worked — the incoming child correctly sat
at `opacity: 0` (via the existing `waitForExit()` gate in `use-visual-element.svelte.ts`). Only its
**layout participation** was wrong.

## The fix

Two changes in `render/dom/motion-outro.ts`:

1. **Hide first, ask afterwards.** `motionEnterIntro` now takes the incoming node out of flow
   imperatively and restores it once `waitForExit()` resolves. This removes the ordering dependency
   completely — we no longer need the outro to have run first. When nothing is exiting,
   `waitForExit()` resolves in the same tick, so a plain enter is restored before paint.

2. **Release a finished exit immediately.** `waitForExit()` resolves when the exit *animation*
   finishes, but Svelte keeps the node mounted until its own outro timer elapses — two different
   clocks. That trailing window still showed both children laid out (`1,1,1,1,2,2,1` in the new
   test). A finished exit now drops out of flow right away; `motionEnterIntro` restores it if the
   exit is reversed before removal.

`mode="sync"` and `mode="popLayout"` are untouched.

## Verification

**New tests** (`components/AnimatePresence/__tests__/wait-mode.svelte.spec.ts`), written RED first:

- never lays out the incoming child while the outgoing one is still exiting
- still ends with exactly the incoming child
- does not hold back the first child when nothing is exiting (no-flash regression guard)

Note the tests assert on **inline** `display`, not CSS keyframes — jsdom does not run CSS
animations, so the original keyframe-based approach was invisible to it. This is part of why the
bug survived.

**Real browser**, `/animate-presence-basics`, sampled every frame across a wait swap:

```
frame  0-19  wait-item-2:none:w=0   ,  wait-item-1:flex:w=200  (opacity 1 -> 0)
frame 28-47  wait-item-2:flex:w=200 (opacity 0 -> 1)
```

At no frame are both children laid out. Before the fix, both were laid out for the whole exit.

**Suite:** 550 unit tests pass (547 before + 3 new), 1 skipped. `sv check` reports 0 errors,
0 warnings.

## Notes

- `context.remaining()` is now unused by the intro. `reserve()`/`remaining()` are left on the outro
  context as-is to keep this change small; they can be retired separately.
- The docs site currently works around this with a single-cell grid on `#content`. That workaround
  should no longer be necessary.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved `AnimatePresence` `mode="wait"` so incoming and outgoing elements don’t share layout during exits.
  * Ensured the incoming element is restored promptly after `waitForExit()` completes, including for SVG and HTML children.
  * Updated outro/wait sequencing to avoid brief layout/visibility edge cases.
  * Confirmed `mode="sync"` and `mode="popLayout"` behavior remains unchanged.

* **Tests**
  * Added/expanded `mode="wait"` fixtures and specs to verify layout visibility and transition ordering, including out-of-flow positioning.

### CODE REVIEW

File: src/lib/motion-start/render/dom/motion-outro.ts [hunk 1]
```diff
@@ -286,6 +287,46 @@ function markProjectionWillUpdate(
...
+
+	hideFromLayout(motionNode);
+
+	const reveal = () => restoreWaitDisplay(motionNode);
+
+	tick()
+		.then(() => context.waitForExit())
+		.then(reveal, reveal);
+}
+
```
[the remark this answers is not in the document]
Comment (JonathonRP) [reply src/lib/motion-start/render/dom/motion-outro.ts:351]: Addressed in d8eadc12. Added an `isWaitDisplayNode()` type guard covering both `HTMLElement` and `SVGElement`, and applied it at all three sites (`hideFromLayout`, `restoreWaitDisplay`, `deferLayoutUntilExitsComplete`) â€” both expose `style` and `dataset`, which is all the hide/restore pair needs.

`WaitModeFixture` gained a `variant` prop, and there's a new `motion.svg` regression case asserting only one `.page` is ever laid out across the transition, plus that the incoming SVG child ends up back in flow.

One deviation from the suggested diff: I did not extend this to `applyPopLayout`. That's `mode=popLayout`, which is a separate code path from the `mode=wait` serialisation this PR fixes, and it wasn't exercised by the bug â€” extending it would be unvalidated scope creep. Happy to do it as a follow-up if you want popLayout SVG support tracked.

Separately, while validating this I found the hide broke `layout-shared.ts`'s `queueMicrotask` case: the fixture's `layoutId` child is `position: absolute`, and a `display: none` node reports a zeroed box, so `onLayoutMeasure` read `x.min === 0`. `hideFromLayout` now skips out-of-flow children, which can't share flow with a sibling anyway. That spec is green again (28/28).

File: src/lib/motion-start/render/dom/motion-outro.ts [hunk 2]
```diff
@@ -286,6 +287,46 @@ function markProjectionWillUpdate(
...
+	motionNode.style.display = 'none';
+}
+
+function restoreWaitDisplay(motionNode: Element) {
+	if (!(motionNode instanceof HTMLElement)) return;
+	const previous = motionNode.dataset.motionWaitDisplay;
+	if (previous === undefined) return;
+	delete motionNode.dataset.motionWaitDisplay;
+	motionNode.style.display = previous;
+}
```
[the remark this answers is not in the document]
Comment (JonathonRP) [reply src/lib/motion-start/render/dom/motion-outro.ts:323]: Addressed in d8eadc12. Added an `isWaitDisplayNode()` type guard covering both `HTMLElement` and `SVGElement`, and applied it at all three sites (`hideFromLayout`, `restoreWaitDisplay`, `deferLayoutUntilExitsComplete`) â€” both expose `style` and `dataset`, which is all the hide/restore pair needs.

`WaitModeFixture` gained a `variant` prop, and there's a new `motion.svg` regression case asserting only one `.page` is ever laid out across the transition, plus that the incoming SVG child ends up back in flow.

One deviation from the suggested diff: I did not extend this to `applyPopLayout`. That's `mode=popLayout`, which is a separate code path from the `mode=wait` serialisation this PR fixes, and it wasn't exercised by the bug â€” extending it would be unvalidated scope creep. Happy to do it as a follow-up if you want popLayout SVG support tracked.

Separately, while validating this I found the hide broke `layout-shared.ts`'s `queueMicrotask` case: the fixture's `layoutId` child is `position: absolute`, and a `display: none` node reports a zeroed box, so `onLayoutMeasure` read `x.min === 0`. `hideFromLayout` now skips out-of-flow children, which can't share flow with a sibling anyway. That spec is green again (28/28).

File: src/lib/motion-start/components/AnimatePresence/__tests__/WaitModeFixture.svelte [hunk 3]
```diff
@@ -0,0 +1,46 @@
+<script lang="ts">
+import { motion } from '../../../index.js';
+import AnimatePresence from '../AnimatePresence.svelte';
+
+let { mode = 'wait' as 'sync' | 'wait' | 'popLayout', variant = 'html' as 'html' | 'svg' | 'absolute' } = $props();
```
[the remark this answers is not in the document]
Comment (JonathonRP) [reply src/lib/motion-start/components/AnimatePresence/__tests__/WaitModeFixture.svelte:7]: Added in 53501418 — this matches the convention used by every other Svelte component in the library (`AnimatePresence.svelte`, `PopChild.svelte`, `MotionScope.svelte` etc all declare it), so relying on inference here was an inconsistency.`sv check` is clean and the AnimatePresence specs still pass (10/10).Note the Biome `noUnusedVariables` warnings on `mode`/`variant` in your tooling output are false positives — both are consumed in the template below, which Biome's Svelte support doesn't track.
